homeassistant/components/generic_thermostat/climate.py

```diff
@@ -521,6 +523,9 @@ async def _async_heater_turn_off(self):
 
     async def async_set_preset_mode(self, preset_mode: str):
         """Set new preset mode."""
+        preset_mode = preset_mode.lower()
+        if preset_mode not in SUPPORTED_PRESETS:
```

Comment:
Instead of this, just add an `else` after the elif.

Reply:
I started with `elif` but changed it to this to be consistent with other code in HA. When I searched for ValueError I saw most methods validate their arguments first and then perform their actions.

Reply:
Also, `lse` doesn't always work because the it would raise when the value didn't change 
`if preset_mode == PRESET_AWAY and not self._is_away`